src/openrct2/paint/Paint.cpp

```diff
@@ -647,36 +649,31 @@ static void PaintPSImageWithBoundingBoxes(rct_drawpixelinfo* dpi, paint_struct*
     gfx_draw_line(dpi, { screenCoordFrontTop, screenCoordRightTop }, colour);
 }
 
-static void PaintPSImage(rct_drawpixelinfo* dpi, paint_struct* ps, uint32_t imageId, int32_t x, int32_t y)
+static void PaintPSImage(rct_drawpixelinfo* dpi, paint_struct* ps, ImageId imageId, int32_t x, int32_t y)
 {
     if (ps->flags & PAINT_STRUCT_FLAG_IS_MASKED)
     {
-        return gfx_draw_sprite_raw_masked(dpi, { x, y }, imageId, ps->colour_image_id);
+        return gfx_draw_sprite_raw_masked(dpi, { x, y }, imageId, ImageId::FromUInt32(ps->colour_image_id));
     }
 
-    gfx_draw_sprite(dpi, imageId, { x, y }, ps->tertiary_colour);
```

Comment:
It seems as if calls to PaintPSImage() have not been modified to put `ps->tertiary_colour` into the ImageId.

Reply:
There are only 3 calls to `PaintPSImage`, one is for painting bound boxes which doesn't need to. The other two are directly after calls to PaintPSColourifyImage.